src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -306,63 +306,97 @@ def rsub(
         """
         return self.__apply_op(y, "sub")
 
-    def mul(self, y):
+    def mul(self, y: Union[int, float]) -> "ReplicatedSharedTensor":
         """Apply the "mul" operation between "self" and "y".
 
         Args:
             y: self*y
 
+        Returns:
+            ReplicatedSharedTensor: Result of the operation.
+
         """
+        y, other = self.sanity_checks(self, y)
+
+        shares = []
+
+        for share in self.shares:
+            shares.append(share * y.shares[0])
+
+        result = ReplicatedSharedTensor(
+            ring_size=self.ring_size, session_uuid=self.session_uuid, config=self.config
+        )
+        result.shares = shares
+
+        return result
 
     def truediv(self, y):
         """Apply the "div" operation between "self" and "y".
 
         Args:
             y: self/y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
     def matmul(self, y):
         """Apply the "matmul" operation between "self" and "y".
 
         Args:
             y: self@y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
     def rmatmul(self, y):
         """Apply the "rmatmul" operation between "y" and "self".
 
         Args:
             y: self@y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
     def xor(self, y):
         """Apply the "xor" operation between "self" and "y".
 
         Args:
             y: self^y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
     def lt(self, y):
         """Lower than operator.
 
         Args:
             y: self<y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
     def gt(self, y):
         """Greater than operator.
 
         Args:
             y: self>y
 
+        Raises:
+            NotImplementedError: Raised when implementation not present
         """
+        raise NotImplementedError
 
-    def eq(self, y: Any):
+    def eq(self, y: Any) -> bool:
```

Comment:
Q: Are we supposed to reveal that if the value under the RSTensors is equal to y without calling a reconstruction? 
that would be a security lapse right?

Reply:
We haven't implemented this method so, can't say what lies in the future or if it will leak anything.

Reply:
I will remove that type annotation. Its for future reference.

Reply:
Oh, I am sorry for the confusion. I thought it was not implemented.

Reply:
hm yes it should be i guess RSTensor since we don't want to reveal any bool until we call reconstruct

Reply:
No, the value can't be got until get() is called. Very similar to how https://github.com/OpenMined/SyMPC/blob/2005f3393c03296bbdf18cecafc9c0582783e84f/src/sympc/tensor/share_tensor.py#L386 has it.